server/authz/rbac.go

```diff
@@ -45,24 +57,75 @@ func NewCasbinObject() (*CasbinObject, error) {
+	policyDefault = getDefaultPolicy(configReader)
+
 	return &CasbinObject{
 		enforcer: enforcer,
 	}, nil
 }
 
-func (cas *CasbinObject) Authorize(c *gin.Context, groups []string) (bool, error) {
+// Authorize checks if a user is authorized to access the resource.
+// It returns true if the user is authorized, otherwise false.
+// It also returns the policy count of the user. The policy count is used to check if there are any policies defined
+// for the given user, if not we will allocate a default policy for the user.
+func (cas *CasbinObject) Authorize(c *gin.Context, userIdentityToken *authn.UserInfo) bool {
+	// Get the scopes to check from the policy.
+	scopedList := getSubjectFromScope(currentScopes, userIdentityToken)
+	// Get the resource, object and action from the request.
 	resource := extractResource(c)
 	object := extractObject(c)
 	action := c.Request.Method
-	// Check if the user has permission for any of the groups.
-	for _, group := range groups {
-		// Get the user from the group. The group is in the format "group:role".
-		// Check if the user has permission using Casbin Enforcer.
-		if ok, _ := cas.enforcer.Enforce(group, resource, object, action); ok {
-			return true, nil
+	userHasPolicies := false
+	// Check for the given scoped list if the user is authorized using any of the subjects in the list.
+	for _, scopedSubject := range scopedList {
+		// Check if the user has permissions in the policy for the given scoped subject.
+		userHasPolicies = userHasPolicies || getPermissionCount(cas.enforcer, scopedSubject)
+		ok := enforceCheck(cas.enforcer, scopedSubject, resource, object, action)
+		if ok {
+			return ok
 		}
 	}
-	return false, fmt.Errorf("user is not authorized to execute the requested action")
+	// If the user does not have any policy defined, allocate a default policy for the user.
+	if !userHasPolicies {
+		logger.Infow("No policy defined for the user, allocating default policy",
+			"DefaultPolicy", policyDefault)
+		ok := enforceCheck(cas.enforcer, policyDefault, resource, object, action)
+		if ok {
+			return ok
+		}
+	}
+	return false
+}
+
+// getSubjectFromScope returns the subjects in the request for the given scopes.
+// The scopes are the params used to check the authentication params to check if the
+// user is authorized to access the resource. For any new scope, add the scope to the
+// rbac properties file and add the scope to the cases below.
+func getSubjectFromScope(scopes []string, userIdentityToken *authn.UserInfo) []string {
+	var scopedList []string
+	// If the scope is group, fetch the groups from the user identity token.
+	if slices.Contains(scopes, ScopeGroup) {
+		scopedList = append(scopedList, userIdentityToken.IDTokenClaims.Groups...)
+	}
+	// If the scope is email, fetch the email from the user identity token.
+	if slices.Contains(scopes, ScopeEmail) {
+		scopedList = append(scopedList, userIdentityToken.IDTokenClaims.Email)
+	}
+	return scopedList
```

Comment:
What if a group is named "abc@gmail.com"?

Reply:
If there is a policy added for `"[abc@gmail.com](mailto:abc@gmail.com)"` then it can work.
So the difference from the authorization side is which param to read from the authentication token. We do not add any type enforcing